starter-code/functions-and-arrays.js

```diff
@@ -1,101 +1,143 @@
 // Find the maximum
-function maxOfTwoNumbers (first, second) {
-
+function maxOfTwoNumbers(first, second) {
+    if (first > second) {
+        return first;
+    } else {
+        return second;
+    }
 }
 
 var largest = maxOfTwoNumbers(2, 6);
 console.log(largest);
 
 // Finding Longest Word
-function findLongestWord (words) {
-
+function findLongestWord(words) {
+    words.forEach(function(words) {
+        if (words.length > lngth) {
+            lngth = words.length;
+            longst = words;
+        }
+    });
+    return longst;
 }
 
 var words = [
-  "mystery",
-  "brother",
-  "aviator",
-  "crocodile",
-  "pearl",
-  "orchard",
-  "crackpot"
+    "mystery",
+    "brother",
+    "aviator",
+    "crocodile",
+    "pearl",
+    "orchard",
+    "crackpot"
 ];
+
+var lngth = 0;
+var longst;
+
 var longest = findLongestWord(words);
 console.log(longest);
 
 // Calculating a Sum
-function sumArray (array) {
-
+function sumArray(array) {
+    var sum = 0;
+    array.forEach(function(array) {
+        sum += array;
+    });
+    return sum;
 }
 
 var numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 var total = sumArray(numbers);
 console.log(total);
 
 // Calculate the Average
-function averageNumbers (array) {
+function averageNumbers(array) {
+    return sum(array) / array.length;
+}
 
+function sum(array) {
+    var sum = 0;
+    array.forEach(function(array) {
+        sum += array;
+    });
+    return sum;
 }
 
 var numbers = [2, 6, 9, 10, 7, 4, 1, 9];
 var average = averageNumbers(numbers);
 console.log(average);
 
 // Array of Strings
-function averageWordLength (array) {
+function averageWordLength(array) {
+    return avg(array) / array.length;
+}
 
+function avg(array) {
+    var avg = 0;
+    array.forEach(function(array) {
+        avg += array.length;
+    });
+    return avg;
 }
 
 var words = [
-  "seat",
-  "correspond",
-  "linen",
-  "motif",
-  "hole",
-  "smell",
-  "smart",
-  "chaos",
-  "fuel",
-  "palace"
+    "seat",
+    "correspond",
+    "linen",
+    "motif",
+    "hole",
+    "smell",
+    "smart",
+    "chaos",
+    "fuel",
+    "palace"
 ];
 var averageLength = averageWordLength(words);
 console.log(averageLength);
 
 // Unique Arrays
-function uniquifyArray (array) {
+function uniquifyArray(array) {
```

Comment:
Buen trabajo!!! Seguid así!!! No olvidéis probar indexOf.. que puede ser util...